energy_analyzer/test_biz_logic.py

```diff
@@ -3,123 +3,118 @@
                     get_first_moment_of_month(measure.start) <= ts < measure.end
                     for measure in building.measures
                 ):
-                    self.assertLess(with_measures["value"], without_measures["value"])
+                    self.assertLess(with_measures.value, without_measures.value)
                 else:
-                    self.assertEqual(with_measures["value"], without_measures["value"])
+                    self.assertEqual(with_measures.value, without_measures.value)
 
     # this should pass once the Challenge task has been completed correctly
     def test_partial_month_coverage(self):
-        start_of_month = get_first_moment_of_month(datetime.now())
-
-        building_partial_coverage = Building(
-            name="Building Partial Coverage",
-            measures=[
-                Measure(
-                    name="Measure Partial Coverage",
-                    measure_type=MeasureType.SAT_RESET,
-                    start=datetime(
-                        year=start_of_month.year,
-                        month=start_of_month.month,
-                        day=6,
-                    ),
-                    end=datetime(
-                        year=start_of_month.year,
-                        month=start_of_month.month,
-                        day=15,
-                    ),
-                )
-            ],
+        result_partial_coverage = building_partial_month_coverage.get_past_and_future_year_of_monthly_energy_usage(
+            include_measure_savings=True
         )
-
-        building_full_coverage = Building(
-            name="Building Full Coverage",
-            measures=[
-                Measure(
-                    name="Measure Full Coverage",
-                    measure_type=MeasureType.SAT_RESET,
-                    start=start_of_month,
-                    end=start_of_month + relativedelta(months=1),
-                )
-            ],
-        )
-
-        result_partial_coverage = (
-            building_partial_coverage.get_past_and_future_year_of_monthly_energy_usage_with_measures()
-        )
-        result_full_coverage = (
-            building_full_coverage.get_past_and_future_year_of_monthly_energy_usage_with_measures()
+        result_full_coverage = building_full_month_coverage.get_past_and_future_year_of_monthly_energy_usage(
+            include_measure_savings=True
         )
 
         for partial_coverage, full_coverage in zip(
             result_partial_coverage, result_full_coverage
         ):
-            ts = partial_coverage["timestamp"]
+            ts = partial_coverage.timestamp
             if ts.year == start_of_month.year and ts.month == start_of_month.month:
-                self.assertLess(full_coverage["value"], partial_coverage["value"])
+                self.assertLess(full_coverage.value, partial_coverage.value)
+
+    def test_get_measure_savings_for_date_range(self):
+        # write this test for the at-home challenge
```

Comment:
You should make it clear whether this needs to be filled in as part of the take home or not.